### PR TITLE
Dependency graphs for types and values

### PR DESCRIPTION
**Pull request by klahnunya**
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: cli/src/Morphir/Web/DevelopApp.elm [hunk 1]
```diff
@@ -1788,7 +1861,16 @@ httpMakeModel =
...
+                            case Repo.fromDistribution result of
+                                Ok repo ->
+                                    ServerGetIRResponse ( result, repo )
+
+                                Err error ->
+                                    let
+                                        _ =
+                                            Debug.log "e" error
+                                    in
+                                    HttpError (Http.BadBody "Could not transform Distribution to Repo")
```
Comment (AttilaMihaly) [inline cli/src/Morphir/Web/DevelopApp.elm:1873]: You should move this bit of code into the `update` function. Generating an HTTP error doesn't seem to be right. The `update` function is the one that should create the repo and update the model to put the application in an errored state if the operation fails. In general any substantial processing should happen within the update function.

File: cli/web/graph.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    </style>
+    
+    <script type="text/javascript" src="elm-graph.js"></script>
+</head>
+
+<body>
+
+<vis-graph id="mynetwork"
+           graph='{"nodes":[{"id": 1, "label": "A"},{"id": 3, "label": "B"}] , "edges": [{"from": 1, "to": 3}]}'>
+</vis-graph>
```
Comment (AttilaMihaly) [inline cli/web/graph.html:21]: This looks like an example. Can you remove it?

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: cli/web/graph.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                smooth: true,
+                arrows: "to",
+            },
+            autoResize: false,
+            height: "1000px",
+            width: "1000px",
+            nodes: {
+                physics: false
+            },
+            title: "FQName"
```
[the remark this answers is not in the document]
Comment (klahnunya) [reply cli/web/graph.js:49]: Yes , this was a temporal placeholder

File: cli/web/graph.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    edgeMinimization: true
+                },
+            },
+            edges: {
+                smooth: true,
+                arrows: "to",
+            },
+            autoResize: false,
+            height: "1000px",
+            width: "1000px",
```
[the remark this answers is not in the document]
Comment (klahnunya) [reply cli/web/graph.js:45]: Yeah i think that is possible

**Comment by AttilaMihaly** [comment]
Merged in a different PR

**Pull request closed without merging**